TransbankPosSDK/POSAutoservicio.cs

```diff
@@ -0,0 +1,184 @@
+            }
+            catch (Exception e)
+            {
+                throw new TransbankException($"Unable to send Poll command on port {Port.PortName}", e);
+            }
+        }
+
+        public async Task<LoadKeysResponse> LoadKeys()
+        {
+            try
+            {
+                await WriteData("0800");
+                return new LoadKeysResponse(CurrentResponse);
+            }
+            catch (Exception e)
+            {
+                throw new TransbankLoadKeysException("Unable to execute Load Keys in pos", e);
+            }
+        }
+
+        public async Task<bool> Initialization()
+        {
+            try
+            {
+                byte[] buffer = new byte[1];
+                string command = "0070";
+
+                Console.WriteLine($"Out (Hex): {ToHexString(command)}");
+                Console.WriteLine($"Out (ASCII): {command}");
+
+                Port.Write(command);
+                await Port.BaseStream.ReadAsync(buffer, 0, 1);
+                return CheckACK(buffer[0]);
+            }
+            catch (Exception e)
+            {
+                throw new TransbankException($"Unable to send Initialization command on port {Port.PortName}", e);
+            }
+        }
+
+        public async Task<InitializationResponse> InitializationResponse()
+        {
+            try
+            {
+                await WriteData("0080\x0B");
+                return new InitializationResponse(CurrentResponse);
+            }
+            catch (Exception e)
+            {
+                throw new TransbankInitializationResponseException("Unable to execute Initialization Response in pos", e);
+            }
+        }
+
+        public async Task<SaleResponse> Sale(int amount, string ticket, bool sendVoucher = false, bool sendStatus = false)
+        {
+            if (amount <= 0)
+            {
+                throw new TransbankSaleException("Amount must be greater than 0");
+            }
+            if (ticket.Length != 6)
+            {
+                throw new TransbankSaleException("Ticket must be 6 characters.");
+            }
+            string message = $"0200|{amount}|{ticket}|{Convert.ToInt32(sendVoucher)}|{Convert.ToInt32(sendStatus)}";
+            try
+            {
+                await WriteData(MessageWithLRC(message), intermediateMessages: sendStatus);
+                return new SaleResponse(CurrentResponse);
+            }
+            catch (Exception e)
+            {
+                throw new TransbankSaleException($"Unable to execute sale on pos", e);
+            }
+        }
+
+        public async Task<MultiCodeSaleResponse> MultiCodeSale(int amount, string ticket, long commerceCode = 0, bool sendVoucher = false, bool sendStatus = false)
+        {
+            if (amount <= 0)
+            {
+                throw new TransbankMultiCodeSaleException("Amount must be greater than 0");
+            }
```

Comment:
There is a max amount also, so you should check for that or the command message length would exceed the max charactaer reserved for that field.

Reply:
Agree

Reply:
the field supports a value of length 9, in theory the maximum amount is 999,999,999

Reply:
😱